subprojects/packagefiles/type_safe/meson.build

```diff
@@ -0,0 +1,57 @@
+project(
+    'type_safe',
+    'cpp',
+    license: 'MIT',
+    version: '0.2.3',
+    meson_version: '>= 0.59.0',
+)
+
+da_dep = dependency('debug_assert')
+
+cxx = meson.get_compiler('cpp')
+
+ndebug_defined = cxx.run('''
+#ifdef NDEBUG
+#define RETURN_MAIN 1
+#else
+#define RETURN_MAIN 0
+#endif
+
+int main() {
+    return RETURN_MAIN;
+}
+''').returncode() == 1
```

Comment:
There's the possible `if-release` value, so I figured it would make more sense to check whether Meson is actually producing `NDEBUG` than try to replicate its logic.